app/Http/Controllers/MessageController.php

```diff
@@ -42,4 +54,36 @@ public function store(Request $request)
             'ids' => $ids ?? null
         ]);
     }
+
+    public function sendDsa(Request $request){
+        $roles = $request['roles'];
+        $staffIDs = [];
+        foreach($roles as $role){
+            $staffs = User::where('role_id', $role)->get()->toArray();
```

Comment:
when getting user, select only phone number and also add where portal access is 1

Reply:
done